Frends.AmazonS3.DownloadObject/Frends.AmazonS3.DownloadObject/DownloadObject.cs

```diff
@@ -11,216 +11,179 @@
     {
-        string responseBody;
-
-        var request = new GetObjectRequest
-        {
-            BucketName = input.BucketName,
-            Key = fileObject.Key
-        };
-
-        using (var response = await s3Client.GetObjectAsync(request))
-        using (var responseStream = response.ResponseStream)
-
-
-        using (var reader = new StreamReader(responseStream)) responseBody = await reader.ReadToEndAsync();
-        if (File.Exists(fullPath))
+        try
         {
-            var file = new FileInfo(fullPath);
+            var retries = 0;
 
-            while (IsFileLocked(file)) Thread.Sleep(1000);
-            File.Delete(fullPath);
-            File.WriteAllText(fullPath, responseBody);
-            return $@"Download with overwrite complete: {fullPath}";
+            while (retries <= fileLockedRetries)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                using FileStream inputStream = File.Open(fullPath, FileMode.Open, FileAccess.Read, FileShare.None);
+                if (inputStream.Length > 0)
+                    return false;
+                else
+                {
+                    retries++;
+                    Thread.Sleep(1000);
+                }
+            }
+            
+            throw new Exception($"FileLocked error: {fullPath} was locked. Max Input.FileLockedRetries = {fileLockedRetries} exceeded.");
         }
-        else
+        catch (Exception ex)
         {
-            if (!Directory.Exists(input.DestinationDirectory)) Directory.CreateDirectory(input.DestinationDirectory);
-            File.WriteAllText(fullPath, responseBody);
-            return $@"Download complete: {fullPath}";
+            throw new Exception($"FileLocked error: {ex}");
         }
     }
 
-    private static async Task<string> DeleteSourceFile(AmazonS3Client client, string bucketName, string key, CancellationToken cancellationToken)
+    private static async Task<bool> DeleteSourceFile(AmazonS3Client client, string bucketName, string key, CancellationToken cancellationToken)
     {
         try
         {
-            var deleteObjectRequest = new DeleteObjectRequest //delete source file from S3
+            var deleteObjectRequest = new DeleteObjectRequest
             {
                 BucketName = bucketName,
                 Key = key
             };
 
-            await client.DeleteObjectAsync(deleteObjectRequest, cancellationToken);
-            return $@"Source file {key} deleted.";
-        }
-        catch (Exception ex) { throw new Exception($"Delete failed. {ex}"); }
-    }
-
-    private static bool IsFileLocked(FileInfo file)
-    {
-        FileStream stream = null;
-
-        try
-        {
-            stream = file.Open(FileMode.Open, FileAccess.ReadWrite, FileShare.None);
+            var deleted = await client.DeleteObjectAsync(deleteObjectRequest, cancellationToken);
```

Comment:
You are not using `deleted`-variable, so you can omit what the method call returns.